File: src/Sharp-O-Matic.Engine/Connections/ConnectionConfig.cs

namespace SharpOMatic.Engine.Connections;

public class ConnectionConfig
{
    public required string ConfigId { get; set; }
    public required string DisplayName { get; set; }
    public required string Description { get; set; }
    public required List<AuthenticationModeConfig> AuthModes { get; set; }
}

File: src/Sharp-O-Matic.Editor/src/app/metadata/definitions/connection-config.ts
import { AuthenticationModeConfig, AuthenticationModeConfigSnapshot } from "./authentication-mode-config";

export interface ConnectionConfigSnapshot {
  id: string;
  displayName: string;
  description: string;
  authModes: AuthenticationModeConfigSnapshot[];
}

export class ConnectionConfig {
  constructor(
    public readonly id: string,
    public readonly displayName: string,
    public readonly description: string,
    public readonly authModes: AuthenticationModeConfig[],
  ) {}

  public static fromSnapshot(snapshot: ConnectionConfigSnapshot): ConnectionConfig {
    return new ConnectionConfig(
      snapshot.id,
      snapshot.displayName,
      snapshot.description,
      snapshot.authModes.map(AuthenticationModeConfig.fromSnapshot),
    );
  }
}
